lib/jormungandr/test/integration/Test/Integration/Jormungandr/Scenario/API/StakePools.hs

```diff
@@ -404,6 +414,79 @@ spec = do
         r <- joinStakePool ctx (p ^. #id) (w, "Secure Passprase")
         expectResponseCode HTTP.status404 r
 
+    it "STAKE_POOLS_ESTIMATE_FEE_01 - \
+        \wallet with [fee - 1] => cannot estimate fee" $ \ctx -> do
+        putStrLn "hello"
+        (_, p:_) <- eventually $
+            unsafeRequest @[ApiStakePool] ctx listStakePoolsEp Empty
+        fee <- do
+            w <- fixtureWallet ctx
+            (_, r) <- joinStakePoolFee ctx (p ^. #id) w
+            print r
+            return $ either (error . show) (getQuantity . amount) r
+
+        w <- fixtureWalletWith ctx [fee - 1]
+        joinStakePoolFee ctx (p ^. #id) w >>= flip verify
+            [ expectResponseCode HTTP.status403
+            , expectErrorMessage "I'm unable to select enough coins to pay for a"
+            ]
+
+    it "STAKE_POOLS_ESTIMATE_FEE/JOIN_01 - \
+        \wallet with [fee] => can join" $ \ctx -> do
+        (_, p:_) <- eventually $
+            unsafeRequest @[ApiStakePool] ctx listStakePoolsEp Empty
+        fee <- do
+            w <- fixtureWallet ctx
+            (_, r) <- joinStakePoolFee ctx (p ^. #id) w
+            return $ either (error . show) (getQuantity . amount) r
+
+        w <- fixtureWalletWith ctx [fee + 1]
+        r <- joinStakePool ctx (p ^. #id) (w, "Secure Passphrase")
+        verify r
+            [ expectResponseCode HTTP.status200
+            , expectFieldEqual status Pending
+            , expectFieldEqual direction Outgoing
+            ]
+        -- Wait for the certificate to be inserted
+        eventually $ do
+            let ep = listTxEp w mempty
+            request @[ApiTransaction n] ctx ep Default Empty >>= flip verify
+                [ expectListItemFieldEqual 0 direction Outgoing
+                , expectListItemFieldEqual 0 status InLedger
+                ]
+
+        -- Verify the wallet is now delegating
+        request @ApiWallet ctx (getWalletEp w) Default Empty >>= flip verify
+            [ expectFieldEqual delegation (Delegating (p ^. #id))
+            ]
```

Comment:
**[question]** 
Why joining a stake pool after looking for the fee? Shouldn't this maybe check that the actual fees are somewhat closed to the estimated one? (they should be exact for Jörmungandr though..).

Reply:
This was testing that a `fixtureWalletWith [fee]` coin join, **for sure**. It found the [edge-case bug]() in the coin-selection when the fee-coefficient was enabled.

But you have a point, it is not relevant to fee estimation; I have removed removed it from this pr.